src/react/reconcilation.js

```diff
@@ -904,6 +904,45 @@ export class Reconciler {
     return value;
   }
 
+  _resolveAbstractLogicalValue(
+    componentType: Value,
+    value: AbstractValue,
+    context: ObjectValue | AbstractObjectValue,
+    evaluatedNode: ReactEvaluatedNode
+  ) {
+    let [leftValue, rightValue] = value.args;
+    let operator = value.kind;
+
+    invariant(leftValue instanceof AbstractValue);
+    if (operator === "||") {
+      return this._resolveAbstractConditionalValue(
+        componentType,
+        leftValue,
```

Comment:
You only want to evaluate leftValue once. This code does not seem quite correct.

Reply:
I was under the impression that `leftValue` only evaluated once with this code. Please can you explain what makes it evaluate twice?

Reply:
Never mind, I was thinking of AST nodes, not a-temporal abstract values.

Reply:
I do, however, wonder if you want to resolve rightValue deeply. If would not be evaluated at runtime if leftValue is truthy.

Reply:
We should be wrapping the right value in a conditional so at runtime it won’t happen unless the left value is truthy